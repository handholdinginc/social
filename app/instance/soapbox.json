{
    "logo": "/instance/hh-logo.svg",
    "navlinks": {
        "homeFooter": [
            { "title": "About", "url": "/about" },
            { "title": "Terms of Service", "url": "/about/tos" },
            { "title": "Privacy Policy", "url": "/about/privacy" },
            { "title": "DMCA", "url": "/about/dmca" }
        ]
    },
    "promoPanel": {
        "items": [{
            "icon": "comment-o",
            "text": "Bloat FE",
            "url": "https://bloat.handholding.io"
        }]
    },
    "extensions": {
        "patron": false
    },
    "defaultSettings": {
        "autoPlayGif": false,
        "theme": "aquatic"
    },
    "copyright": "Copyright 2022 Handholding Social.",
    "brandColor": "#ff7139"
}
