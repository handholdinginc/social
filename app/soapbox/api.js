'use strict';

import axios from 'axios';
import LinkHeader from 'http-link-header';
import { getAccessToken, getAppToken } from 'soapbox/utils/auth';

export const getLinks = response => {
  const value = response.headers.link;
  if (!value) return { refs: [] };
  return LinkHeader.parse(value);
};

const getToken = (getState, authType) => {
  const state = getState();
  return authType === 'app' ? getAppToken(state) : getAccessToken(state);
};

const maybeParseJSON = data => {
  try {
    return JSON.parse(data);
  } catch(Exception) {
    return data;
  }
};

export const baseClient = accessToken => {
  return axios.create({
    headers: Object.assign(accessToken ? {
      'Authorization': `Bearer ${accessToken}`,
    } : {}),

    transformResponse: [maybeParseJSON],
  });
};

export default (getState, authType = 'user') => {
  const accessToken = getToken(getState, authType);
  return baseClient(accessToken);
};
