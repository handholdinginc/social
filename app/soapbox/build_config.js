// @preval
/**
 * Build config: configuration set at build time.
 * @module soapbox/build_config
 */

const { trim } = require('lodash');

const {
  BACKEND_URL,
  FE_SUBDIRECTORY,
  FE_BUILD_DIR,
} = process.env;

const sanitizeURL = url => {
  try {
    return new URL(url).toString();
  } catch {
    return '';
  }
};

const sanitizeBasename = path => {
  return `/${trim(path, '/')}`;
};

const sanitizePath = path => {
  return trim(path, '/');
};

// JSON.parse/stringify is to emulate what @preval is doing and avoid any
// inconsistent behavior in dev mode
const sanitize = obj => JSON.parse(JSON.stringify(obj));

module.exports = sanitize({
  BACKEND_URL: sanitizeURL(BACKEND_URL),
  FE_SUBDIRECTORY: sanitizeBasename(FE_SUBDIRECTORY),
  FE_BUILD_DIR: sanitizePath(FE_BUILD_DIR) || 'static',
});
