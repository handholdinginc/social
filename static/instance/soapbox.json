{
    "navlinks": {
        "homeFooter": [
            { "title": "About", "url": "/about" },
            { "title": "Terms of Service", "url": "/about/tos" },
            { "title": "Privacy Policy", "url": "/about/privacy" },
            { "title": "DMCA", "url": "/about/dmca" }
        ]
    },
    "promoPanel": {
        "items": [{
            "icon": "film",
            "text": "Streaming",
            "url": "https://stream.handholding.io"
        }, {
            "icon": "comment-o",
            "text": "Lightweight Frontend (Bloat)",
            "url": "https://bloat.handholding.io"
        }]
    },
    "extensions": {
        "patron": false
    },
    "defaultSettings": {
        "autoPlayGif": false,
        "theme": "aquatic"
    },
    "copyright": "Copyright 2020 Handholding Social.",
    "brandColor": "#f99d00"    
}
