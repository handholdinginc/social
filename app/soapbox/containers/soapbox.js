'use strict';

import React from 'react';
import { Provider, connect } from 'react-redux';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import SoapboxPropTypes from 'soapbox/utils/soapbox_prop_types';
import Helmet from 'soapbox/components/helmet';
import classNames from 'classnames';
import configureStore from '../store/configureStore';
import { INTRODUCTION_VERSION } from '../actions/onboarding';
import { Switch, BrowserRouter, Route } from 'react-router-dom';
import { ScrollContext } from 'react-router-scroll-4';
import UI from '../features/ui';
// import Introduction from '../features/introduction';
import { preload } from '../actions/preload';
import { IntlProvider } from 'react-intl';
import { previewState as previewMediaState } from 'soapbox/features/ui/components/media_modal';
import { previewState as previewVideoState } from 'soapbox/features/ui/components/video_modal';
import ErrorBoundary from '../components/error_boundary';
import { fetchInstance } from 'soapbox/actions/instance';
import { fetchSoapboxConfig } from 'soapbox/actions/soapbox';
import { fetchMe } from 'soapbox/actions/me';
import PublicLayout from 'soapbox/features/public_layout';
import { getSettings } from 'soapbox/actions/settings';
import { getSoapboxConfig } from 'soapbox/actions/soapbox';
import { generateThemeCss } from 'soapbox/utils/theme';
import messages from 'soapbox/locales/messages';
import { FE_SUBDIRECTORY } from 'soapbox/build_config';

const validLocale = locale => Object.keys(messages).includes(locale);

export const store = configureStore();

store.dispatch(preload());

store.dispatch(fetchMe())
  .then(() => {
    // Postpone for authenticated fetch
    store.dispatch(fetchInstance());
    store.dispatch(fetchSoapboxConfig());
  })
  .catch(() => {});

const mapStateToProps = (state) => {
  const me = state.get('me');
  const account = state.getIn(['accounts', me]);
  const showIntroduction = account ? state.getIn(['settings', 'introductionVersion'], 0) < INTRODUCTION_VERSION : false;
  const settings = getSettings(state);
  const soapboxConfig = getSoapboxConfig(state);
  const locale = settings.get('locale');

  return {
    showIntroduction,
    me,
    reduceMotion: settings.get('reduceMotion'),
    underlineLinks: settings.get('underlineLinks'),
    systemFont: settings.get('systemFont'),
    dyslexicFont: settings.get('dyslexicFont'),
    demetricator: settings.get('demetricator'),
    locale: validLocale(locale) ? locale : 'en',
    themeCss: generateThemeCss(soapboxConfig.get('brandColor')),
    brandColor: soapboxConfig.get('brandColor'),
    themeMode: settings.get('themeMode'),
    halloween: settings.get('halloween'),
    customCss: soapboxConfig.get('customCss'),
  };
};

@connect(mapStateToProps)
class SoapboxMount extends React.PureComponent {

  static propTypes = {
    showIntroduction: PropTypes.bool,
    me: SoapboxPropTypes.me,
    reduceMotion: PropTypes.bool,
    underlineLinks: PropTypes.bool,
    systemFont: PropTypes.bool,
    dyslexicFont: PropTypes.bool,
    demetricator: PropTypes.bool,
    locale: PropTypes.string.isRequired,
    themeCss: PropTypes.string,
    themeMode: PropTypes.string,
    brandColor: PropTypes.string,
    customCss: ImmutablePropTypes.list,
    halloween: PropTypes.bool,
    dispatch: PropTypes.func,
  };

  state = {
    messages: {},
    localeLoading: true,
  }

  setMessages = () => {
    messages[this.props.locale]().then(messages => {
      this.setState({ messages, localeLoading: false });
    }).catch(() => {});
  }

  maybeUpdateMessages = prevProps => {
    if (this.props.locale !== prevProps.locale) {
      this.setMessages();
    }
  }

  componentDidMount() {
    this.setMessages();
  }

  componentDidUpdate(prevProps) {
    this.maybeUpdateMessages(prevProps);
  }

  shouldUpdateScroll(_, { location }) {
    return location.state !== previewMediaState && location.state !== previewVideoState;
  }

  render() {
    const { me, themeCss, locale, customCss } = this.props;
    if (me === null) return null;
    if (this.state.localeLoading) return null;

    // Disabling introduction for launch
    // const { showIntroduction } = this.props;
    //
    // if (showIntroduction) {
    //   return <Introduction />;
    // }

    const bodyClass = classNames('app-body', `theme-mode-${this.props.themeMode}`, {
      'system-font': this.props.systemFont,
      'no-reduce-motion': !this.props.reduceMotion,
      'underline-links': this.props.underlineLinks,
      'dyslexic': this.props.dyslexicFont,
      'demetricator': this.props.demetricator,
      'halloween': this.props.halloween,
    });

    return (
      <IntlProvider locale={locale} messages={this.state.messages}>
        <ErrorBoundary>
          <Helmet>
            <body className={bodyClass} />
            {themeCss && <style id='theme' type='text/css'>{`:root{${themeCss}}`}</style>}
            {customCss && customCss.map(css => (
              <link rel='stylesheet' href={css} key={css} />
            ))}
            <meta name='theme-color' content={this.props.brandColor} />
          </Helmet>
          <BrowserRouter basename={FE_SUBDIRECTORY}>
            <ScrollContext shouldUpdateScroll={this.shouldUpdateScroll}>
              <Switch>
                {!me && <Route exact path='/' component={PublicLayout} />}
                <Route exact path='/about/:slug?' component={PublicLayout} />
                <Route path='/' component={UI} />
              </Switch>
            </ScrollContext>
          </BrowserRouter>
        </ErrorBoundary>
      </IntlProvider>
    );
  }

}

export default class Soapbox extends React.PureComponent {

  printConsoleWarning = () => {
    /* eslint-disable no-console */
    console.log('%cStop!', [
      'color: #ff0000',
      'display: block',
      'font-family: system-ui, -apple-system, BlinkMacSystemFont, Roboto, Ubuntu, "Helvetica Neue", sans-serif',
      'font-size: 50px',
      'font-weight: 800',
      'padding: 4px 0',
    ].join(';'));
    console.log('%cThis is a browser feature intended for developers. If someone told you to copy-paste something here it is a scam and will give them access to your account.', [
      'color: #111111',
      'display: block',
      'font-family: system-ui, -apple-system, BlinkMacSystemFont, Roboto, Ubuntu, "Helvetica Neue", sans-serif',
      'font-size: 18px',
      'padding: 4px 0 16px',
    ].join(';'));
    /* eslint-enable no-console */
  }

  componentDidMount() {
    this.printConsoleWarning();
  }

  render() {
    return (
      <Provider store={store}>
        <SoapboxMount />
      </Provider>
    );
  }

}
