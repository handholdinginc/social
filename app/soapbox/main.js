'use strict';

import './wdyr';
import './precheck';
// FIXME: Push notifications are temporarily removed
// import * as registerPushNotifications from './actions/push_notifications';
// import { default as Soapbox, store } from './containers/soapbox';
import { default as Soapbox } from './containers/soapbox';
import React from 'react';
import ReactDOM from 'react-dom';
import ready from './ready';

const perf = require('./performance');

function main() {
  perf.start('main()');

  ready(() => {
    const mountNode = document.getElementById('soapbox');

    ReactDOM.render(<Soapbox />, mountNode);
    if (process.env.NODE_ENV === 'production') {
      // avoid offline in dev mode because it's harder to debug
      require('offline-plugin/runtime').install();
      // FIXME: Push notifications are temporarily removed
      // store.dispatch(registerPushNotifications.register());
    }
    perf.stop('main()');
  });
}

export default main;
