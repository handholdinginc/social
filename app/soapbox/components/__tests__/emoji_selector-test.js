import React from 'react';
import { createComponent } from 'soapbox/test_helpers';
import EmojiSelector from '../emoji_selector';

describe('<EmojiSelector />', () => {
  it('renders correctly', () => {
    const component = createComponent(<EmojiSelector />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
