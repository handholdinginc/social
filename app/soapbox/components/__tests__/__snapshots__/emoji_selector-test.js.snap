// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiSelector /> renders correctly 1`] = `
<div
  className="emoji-react-selector-container"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex="-1"
>
  <div
    className="emoji-react-selector"
    onBlur={[Function]}
  >
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"👍\\" title=\\":+1:\\" src=\\"/emoji/1f44d.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"❤\\" title=\\":heart:\\" src=\\"/emoji/2764.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"😆\\" title=\\":laughing:\\" src=\\"/emoji/1f606.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"😮\\" title=\\":open_mouth:\\" src=\\"/emoji/1f62e.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"😢\\" title=\\":cry:\\" src=\\"/emoji/1f622.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
    <button
      className="emoji-react-selector__emoji"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<img draggable=\\"false\\" class=\\"emojione\\" alt=\\"😩\\" title=\\":weary:\\" src=\\"/emoji/1f629.svg\\" />",
        }
      }
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    />
  </div>
</div>
`;
